src/detect-dsize.c

```diff
@@ -896,80 +627,68 @@ static int DsizeTestParse21(void)
 /**
  * \test this is a test for a valid dsize value ! 1
  *
- *  \retval 1 on success
- *  \retval 0 on failure
  */
-static int DsizeTestParse22(void)
+static int DsizeTestParse10(void)
 {
-    DetectDsizeData *dd = NULL;
-    dd = DetectDsizeParse("! 1");
+    DetectDsizeData *dd = DetectDsizeParse("! 1");
     FAIL_IF_NULL(dd);
     DetectDsizeFree(NULL, dd);
     PASS;
 }
 
 /**
- * \test this is a test for a invalid dsize value 1!
+ * \test this is a test for invalid dsize values
+ * A, >10<>10, <>10, 1<>, "", " ", 2<>1, 1!
  *
- *  \retval 1 on success
- *  \retval 0 on failure
  */
-static int DsizeTestParse23(void)
+static int DsizeTestParse11(void)
 {
-    DetectDsizeData *dd = NULL;
-    dd = DetectDsizeParse("1!");
-    if (dd) {
-        DetectDsizeFree(NULL, dd);
-        FAIL;
+    const char *strings[] = { "A", ">10<>10", "<>10", "1<>", "", " ", "2<>1", "1!", NULL };
+    for (int i = 0; strings[i]; i++) {
```

Comment:
Not sure if this format is welcome for tests...

Reply:
I'm OK with these "table driven tests". The only issue is that if it did fail we have no context to show which one failed. But some debug statements could easily help with that when debugging why it failed I guess.

Reply:
I'm good with the table approach. Think its easy enough to find where it fails when it does.

An alternative approach could be something like the macros here  https://github.com/OISF/suricata/blob/master/src/tests/stream-tcp-list.c#L65

Reply:
Oh, I had not crossed those before, thanks!!